disperser/common/blobstore/blob_metadata_store.go

```diff
@@ -59,11 +61,16 @@ func (s *BlobMetadataStore) QueueNewBlobMetadata(ctx context.Context, blobMetada
 	if s.shadowTableName != "" && s.shadowTableName != s.tableName {
 		err = s.dynamoDBClient.PutItem(ctx, s.shadowTableName, item)
 		if err != nil {
+			// TODO: should we return an error here? If not, then we should document why we are ignoring the error
 			s.logger.Error("failed to put item into shadow table %s : %v", s.shadowTableName, err)
 		}
 	}
 
-	return s.dynamoDBClient.PutItem(ctx, s.tableName, item)
+	err = s.dynamoDBClient.PutItem(ctx, s.tableName, item)
+	if err != nil {
+		return api.NewErrorUnavailable(fmt.Sprintf("failed to put item into table %s : %v", s.tableName, err))
```

Comment:
I don't think we should be returning api errors from libraries. 
Maybe we can create new errors at the library level and match those in apiserver to identify which api error should wrap it?

Reply:
Agree. I started working on a generic API error in https://github.com/Layr-Labs/eigenda/pull/828/files#diff-b14f8e128e2a8997e2b4f5f10af13deeba5cc983f7aad6c540227c0908fada4a

It's under api/errors.go right now, since I'm only using it as part of the clients, but perhaps I can move it higher (I'd be in favor of having it in errors.go under the root dir and using such an error across the codebase personally.

WDYT?